diff --git a/working_project/latest_db_row.py b/working_project/latest_db_row.py
--- working_project/latest_db_row.py
+++ working_project/latest_db_row.py
@@ -1,7 +1,8 @@
 from sqlalchemy import text, create_engine
 from sqlalchemy.exc import ProgrammingError
 
+# Everything is `object`. This annotation means nothing. You'd want `datetime | None`.
 def latest_timestamp(engine) -> object:
     """
     Docstring for latest_timestamp
     this function fetchest latest added row of the db table current_forecast
@@ -21,6 +22,7 @@
                 latest = result.fetchone()
                 return latest[0]
         except ProgrammingError as pe:
             print(f'error from {pe}')
+            # It makes more sense to return None, but this technically works.
             return False
         
\ No newline at end of file
diff --git a/main.py b/main.py
--- main.py
+++ main.py
@@ -1,13 +1,26 @@
 import pandas as pd 
 import requests
 from sqlalchemy import create_engine
+# It's good to avoid a single giant file, but it's also cluttery to split each function into their own file.
+# There's good opportunity for grouping here: you have an API function, and a function that processes the raw API response.
+# You have a function to create database tables, you have a function to query the database.
 from working_project.data_cleaning import clean_data
 from working_project.create_db import create_tables
 from working_project.api_calls import call_api
 from working_project.latest_db_row import latest_timestamp
 
+# Commit messages are intended to be formatted like:
+#   A short descriptive title
+#   
+#   Further context or reasoning for the changes
+#
+# You have a commit titled "refactored the multiple files according to class mates suggestions  call_api returns now binary, clean_data function got rid of redundant converting str to binary extra variables df2,df3, create_tables() now receives sqlalchemy engine to connect to db also got rid of redundant conn1 and conn2 now there is one connection and it closes after it gets outside of with created new file latest_db_row it has latest_timestamp it returns datetime object which is used by the main() this allows that duplicates are dealt in the pandas the main.py the_project() is now more clear"
+# Please, press Enter.
 
+# You also committed the .env file into the repo, and only used decouple() in the temporary "main copy.py" you deleted.
+
+
 # 
 def the_project():
     """
     this method stores data to the postgres tables 
@@ -18,27 +31,33 @@
     path_to_db = 'postgresql://myuser:mypassword@localhost:5432/mydatabase'
     engine = create_engine(path_to_db)
 
     resp = call_api()
+    # Don't compare None with ==, just `if not resp:`. (If you have to know whether something *is* specifically None and not just falsey, then `if x is None:`)
     if resp == None:
         raise TypeError('response of the api request is None expected xml')
 
     df = clean_data(resp)
     # fetches most recent timestamp from the db 
+    # The tables might not exist yet, though that's technically handled
     if latest_time := latest_timestamp(engine): 
             df = df[df['timestamps'] > latest_time].reset_index(drop=True)
     print(len(df))
     print('here and all good')
+    # It's generally better to order the "cheaper" clause first (df.empty) and more expensive/mutating ones later (creating tables)
     if create_tables(engine) and not df.empty:
         try:
             df.to_sql('raw_forecast', engine, if_exists='append', index=False, method='multi', chunksize=500)
             df.to_sql('current_forecast', engine, if_exists='append', index=False, method='multi', chunksize=500)
         except Exception as e:
             print('error is likely related to duplicate data in current_forecast table')
             print(f'there was error {e}')
+    # The return value is still not used and communicates nothing (ie, it doesn't report back about whether new data was inserted, it's always True).
+    # Also, it's unused.
     return True
 
 def main():
+    # Still a main() that calls another main-like function and nothing else.
     the_project()
 
 
 if __name__ == "__main__":
diff --git a/working_project/api_calls.py b/working_project/api_calls.py
--- working_project/api_calls.py
+++ working_project/api_calls.py
@@ -14,7 +14,9 @@
 
     resp = requests.get(url=url, params=params)
     resp.raise_for_status()
     print(type(resp.content))
+    # You're returning the bytes, not text. ElementTree still eats text, and only handles bytes as a courtesy.
     return resp.content 
 
+# !!You're calling the api when the module is loaded!!
 call_api()
\ No newline at end of file
